packages/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -279,16 +313,28 @@ export class ConnectionPropsForSessCfg {
         tokenType: SessConstants.TOKEN_TYPE_CHOICES
     ) {
         const impLogger = Logger.getImperativeLogger();
-        let logMsgtext = "Using basic authentication ";
+        let logMsgtext;
+
+        if (sessCfg.cert && sessCfg.certKey) {
+            logMsgtext = "Using PEM certificate authentication ";
+            sessCfg.type = SessConstants.AUTH_TYPE_CERT_PEM;
+        // } else if (sessCfg.cert && sessCfg.passphrase) {
+        //      logMsgtext = "Using PFX certificate authentication ";
+        //      sessCfg.type = SessConstants.AUTH_TYPE_CERT_PFX;
+        } else {
+            logMsgtext = "Using basic authentication ";
+            sessCfg.type = SessConstants.AUTH_TYPE_BASIC;
+        }
 
         if (options.requestToken) {
             // Set our type to token to get a token from user and pass
             logMsgtext += "to get token";
-            sessCfg.type = SessConstants.AUTH_TYPE_TOKEN;
+            if (!sessCfg.cert) {
```

Comment:
As implemented, the session will only work with certificates if the session type is set to `cert-pem`. This was to try and handle some situations that only occur with certificate based authentication (for example, it only being permitted on HTTPS sessions, requiring existing files on disk, requiring the files to be in a certain format, and in the future with PFX certificates, MAC Verification). I suppose it should be possible to duplicate that logic as a conditional for token based sessions, but I will need to implement more tests. I don't believe I was considering the possibility of a session being re-used after a login action.

Reply:
Discussed in person and agreed that this makes sense for now. We have to handle the token specially for cert sessions, since the session type must be "cert-pem", unlike basic auth sessions which can contain user and password even if the session type is "token".